### ISSUE TITLE
[CCR] Follower indices should be resumable without needing to be paused first

### ISSUE THREAD

**Issue by cjcenizal** [elastic/elasticsearch#41373]
CC @martijnvg @jen-huang 

It would be ideal if we could hit `_ccr/resume_follow` without having to first pause the follower index. This would simplify both our client-side and server-side logic in Kibana, and simplify the UX as well since we currently need to inform the user about implementation details when they update a follower index:

![image](https://user-images.githubusercontent.com/1238659/56426853-893e5900-626e-11e9-8a8f-1576ee765399.png)

**Comment by martijnvg** [comment]
@cjcenizal I agree, it would be nicer in general if the advanced settings are changeable without having to pause and resume. We kept the advanced setting immutable for the first version of CCR, but this is something that we can change. I do wonder what setting users end up changing. Ideally these settings should rarely (or never) be changed. The defaults should work well.

**Comment by cjcenizal** [comment]
Thank you @martijnvg! Fortunately we've instrumented the UI to gather some telemetry in 7.1. We'll be able to see how often people are creating follower indices with changed settings, how often they're updating their follower indices, and how often they're updating them with changed settings. Though we won't be able to see which specific settings are being changed. At least these metrics will give us an idea of how much value this change will provide.

**Comment by martijnvg** [comment]
Great to hear that we will get more insight how CCR is used!

**Comment by tlrx** [comment]
@cjcenizal I'm doing a bit of spring cleaning, do you know if this change is still relevant?

**Comment by cjcenizal** [comment]
@tlrx Yes, it's still relevant. I should clarify that this request is in support of our [update follower index API](https://github.com/elastic/kibana/blob/master/x-pack/plugins/cross_cluster_replication/server/routes/api/follower_index/register_update_route.ts). This API consumes the [ES Resume Follow API](https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-post-resume-follow.html) because that's the only way to change a follower index's configuration. However, it also requires an index to be paused or else it will reject the request.

The goal is a simple developer experience (and to a lesser extent, a simpler user experience). I can see two options:
1. Add an "Update Follower Index API" so we can apply changes to a follower index without having to pause and resume it. This would be the best DX, IMO.
2. Implement the original request, which is to support calling the Resume Follower API with an active follower index.
